### DISCUSSION TITLE
Can Rsbuild do SSR (Server side Rendering)?

### DISCUSSION THREAD

**Discussion by NekoSakuraLucia** [web-infra-dev/rsbuild#2881] [General]
I would like to know if Rsbuild can do SSR. 🌙

**Comment by 9aoy** [comment]
Of course, you can refer to this example: [SSR + Express Example](https://github.com/rspack-contrib/rspack-examples/tree/main/rsbuild/ssr-express). 
Rsbuild does not have built-in support for SSR, but Rsbuild server provides a series of APIs related to the build environment and the ability to extend the server.
You can quickly implement an SSR function in development mode through the [Rsbuild environment API](https://rsbuild.dev/guide/advanced/environments#environment-api).
